### DISCUSSION TITLE
Guides and Videos on Data Modeling and Relationships

### DISCUSSION THREAD

**Discussion by bryantgillespie** [directus/directus#18064] [Archive]
Hi folks 👋  – Bryant here - Dev Advocate at Directus...

Starting work on some guides and videos on data modeling and relationships soon.

Would love to have some input from you fine community members you... 🙏 

Few questions for you all.

**1) Any specific topics or questions about data modeling and relationships in Directus that you'd like to see us cover?**

**2) What are the things that trip you up when building your data model in Directus?**

**3) Any specific "sample" data models you'd like to see made available as starting points or quasi-templates?**

**Comment by francisco-telleria** [comment]
Hey, just saw this post, here are my two cents:

1) A general intro to the different types of relationships, and how data is actually stored, like m2m is a separate table... I remember this being covered very well in the docs. Just a refresher. Also, some general guidelines on what relationship to choose with some examples.
2) I haven't used Directus enough but many times I struggle to detect the right relationship, even if I know in my head almost always I have to double check while I'm doing them. I remember strapi has a very obvious UI where it tells you via a phrase and with some icons how the relationship works.
3) Don't have one that I can share but I have a db that I'm planning to move to Directus and in the process we are looking for improvements we can do to the data model, so maybe we can use some examples out of there.

In general, I would make the videos related to Directus but appealing to a big audience of people that's new to data models, this way you help Directus users while also attracting potential new adopters.

Gret initiative, keep it up!

**Reply by pavelstianko** [reply]
Hi Bryant! Same as few members here I used Wordpress with ACF and had a problem to find out how to create the same functionality in Directus :) The most complicated for me were relationships - how to create ACF-like repeater and ACF-like relationship. For my help,  I created this schema:
![image](https://user-images.githubusercontent.com/13132639/231393391-913c57ff-c736-4ad4-90cb-ee9004d67806.png)

It would be nice to have same level of documentation like here: https://www.advancedcustomfields.com/resources/repeater/ . Good explanation of datatype, screenshot or videos and how I can use it in SDK + some basic examples. Thank you for awesome work!

**Reply by bryantgillespie** [reply]
@pavelstianko Nice job on the supporting graphics. I can definitely see how those are super helpful. 

> It would be nice to have same level of documentation like here: https://www.advancedcustomfields.com/resources/repeater/ . Good explanation of datatype, screenshot or videos and how I can use it in SDK + some basic examples. Thank you for awesome work!

I'll definitely review this with the team!

We recently added a "catalog" of all the different fields available which I think is helpful but we can always improve.
For example – https://docs.directus.io/configuration/data-model/fields/relational.html#many-to-many

We also have some supporting graphics in the Relationship section but I just noticed that it doesn't have the directionality / ownership display in the graphic so you have to read between the lines a bit.
https://docs.directus.io/configuration/data-model/relationships.html#many-to-one-m2o
![image](https://user-images.githubusercontent.com/23302570/231450236-b61dc537-3356-4862-906f-90949e185a25.png)

**Comment by bryantgillespie** [comment]
@francisco-telleria Thanks for the feedback. Very helpful!

> I haven't used Directus enough but many times I struggle to detect the right relationship, even if I know in my head almost always I have to double check while I'm doing them. I remember strapi has a very obvious UI where it tells you via a phrase and with some icons how the relationship works.

You're referring to this UI inside Strapi? And the little icons in the center?
![image](https://user-images.githubusercontent.com/23302570/231448651-86a38bc1-76e9-40df-9d47-69f909dda517.png)
